src/reynir_correct/errfinder.py

```diff
@@ -1256,8 +1256,8 @@ def visit_nonterminal(self, level: int, node: Node) -> Any:
                         AnnotationTuple6, ann
                     )
             else:
-                if len(ann) == 0:
-                    # Empty dict: this means that upon closer inspection,
+                if not ann or len(ann) == 0:
```

Comment:
if not ann: is enough (if len(ann) is 0 then it is falsy).

Reply:
Done